### PR TITLE
Abstract strategy gas improvements (#1719)

### PR DESCRIPTION
**Pull request by naddison36**
Contract changes:
* `platformAddress` and `vaultAddress` in the `InitializableAbstractStrategy` changed to be immutables.
* private deprecated variables added to `InitializableAbstractStrategy` to preserve the storage slots.
* `constructor` added to each of the strategy contracts to take the `BaseStrategyConfig` struct that contains `platformAddress` and `vaultAddress`.
* the `initialize` function on the strategy contract changed to not include the `platformAddress` and `vaultAddress` params.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. 
  - [ ] Copy & paste code review [security checklist](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md) below this checklist.
  - [ ] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

File: contracts/contracts/strategies/AaveStrategy.sol [hunk 1]
```diff
@@ -22,21 +22,24 @@ contract AaveStrategy is InitializableAbstractStrategy {
     IAaveIncentivesController public incentivesController;
     IAaveStakedToken public stkAave;
 
+    /**
+     * @param _stratConfig The platform and OToken vault addresses
+     */
+    constructor(BaseStrategyConfig memory _stratConfig)
```
Comment (sparrowDom) [inline contracts/contracts/strategies/AaveStrategy.sol:28]: should all these constructor params be calldata instead of memory?

**Comment by sparrowDom** [comment]
There look like ok changes to me on the other hand probably no need to rush it for the code freeze today?

**Comment by sparrowDom** [comment]
Ah so we are only scoping today and not doing the actual code freeze.

There are some fork tests failing other than that LGTM

**Review by sparrowDom — APPROVED** [review]
LGTM

**Comment by naddison36** [comment]
The fork tests are failing as we have removed liquidity from the strategies. They should work again once we reenable the strategies. If not, I'll fix

File: contracts/contracts/strategies/AaveStrategy.sol (hunk 1 above)
Comment (naddison36) [reply contracts/contracts/strategies/AaveStrategy.sol:28]: only `storage` or `memory` is supported for struct param.

**Pull request merged**